### PR TITLE
Deal with contract destory

### PR DESCRIPTION
**Pull request by bettybao1209**
partly close https://github.com/neo-project/neo-modules/issues/195

### CODE REVIEW

**Comment by erikzhang** [comment]
You shouldn't modify the database when querying.

**Comment by erikzhang** [comment]
https://github.com/neo-project/neo-modules/blob/bbcfa56171a931c1c6f2b50f9289350dd7c052c0/src/RpcNep5Tracker/RpcNep5Tracker.cs#L148

You can delete them in `OnPersist()`.

**Comment by bettybao1209** [comment]
> You can delete them in `OnPersist()`.
But I think it is hard to delete records for non-owner address.

**Comment by nicolegys** [comment]
Thank you @bettybao1209 

Test partly passed.
After the contract being destroyed, getnep5balances won't return its nep5 asset.

Steps:
1.deploy a nep5 contract.
2.make some txs and then getnep5balances.
image
3.destroy the contract.
image
4.getnep5balances
image
The asset was deleted from the list.

Some concerns:
1.Now after the nep5 contract is destroyed, it will not be listed when calling `getnep5balances`, but its storage items still exist.
Once the owner deploys the same contract again, when we call `getnep5balances`, it's OK for the owner address, because the value will be covered, but for the non-owner addresses, the old value will be listed until the address have a `transfer` tx to cover the old value.
I know the probability for someone to re-deploy his destroyed contract is quite low, but maybe we should find a way to delete the storage item in the future.
2.Also, after the nep5 contract is migrated, its "asset_hash" won't be refreshed when calling `getnep5balances`. Every holder address must make a `transfer` tx to refresh its "asset_hash". 
I think we can find a better solution in the future.

File: src/RpcNep5Tracker/RpcNep5Tracker.cs [hunk 1]
```diff
@@ -286,6 +286,8 @@ public JObject GetNep5Balances(JArray _params)
             foreach (var (key, value) in dbCache.Find(prefix))
             {
                 JObject balance = new JObject();
+                if (Blockchain.Singleton.View.Contracts.TryGet(key.AssetScriptHash) is null)
+                    continue;
```
Comment (shargon) [inline src/RpcNep5Tracker/RpcNep5Tracker.cs]: Maybe we shoud do that only in the first iteration, a break or a flag should be ok

Comment (bettybao1209) [reply src/RpcNep5Tracker/RpcNep5Tracker.cs]: But we should list all assets in user's address, so we can't just break the iteration.

**Pull request merged**